txtorcon/torconfig.py

```diff
@@ -423,6 +424,11 @@ class EphemeralHiddenService(object):
     https://gitweb.torproject.org/torspec.git/tree/control-spec.txt#n1295
     '''
 
+    @classmethod
+    def _is_valid_keyblob(cls, key_blob_or_type):
+        return (isinstance(key_blob_or_type, str) and
+                re.match(r'[^ :]+:[^ :]+$', key_blob_or_type))
```

Comment:
Yeah, it does kind of break things, unicode-wise. Maybe what we should do is use the "nativeString" (I think it's called?) helper from Twisted.

The idea here is: if anyone is right now using "`NEW:BEST`" or similar (i.e. non-unicode on Py2) then we can try to upgrade it to unicode by pretending it's ASCII (`blob.decode('ascii')`). If that works, great! Otherwise, no error. Then this works for people using the "meejah didn't think about unicode properly" API already...

So, basically: a helper in `util` that "maybe converts to unicode, or throws an exception" given some input-thing. Or just use `twisted.python.compat.nativeString`...